### PR TITLE
feat(react-grid): disallow unknown column names

### LINKED ISSUE #1495: Auto Grid: type checking for string property references
Some properties of Auto Grid require referencing model properties by string, namely `visibleColumns`, and `columnOptions`.

Currently the TypeScript definitions for those are just `string`, no static type checking is done on the string name.

Expected: TypeScript error for unknown property names that are not keys in the given model.

### PR DESCRIPTION
**Pull request by platosha**
Fixes #1495

### CODE REVIEW

File: packages/ts/react-crud/src/autogrid.tsx [hunk 1]
```diff
@@ -23,16 +23,84 @@ import type PropertyStringFilter from './types/dev/hilla/crud/filter/PropertyStr
 import type Sort from './types/dev/hilla/mappedtypes/Sort';
 import Direction from './types/org/springframework/data/domain/Sort/Direction';
 
+type AllPropertyNames<TItem> = string & keyof TItem;
```
Comment (sissbruecker) [inline packages/ts/react-crud/src/autogrid.tsx:26]: The new types slow down type-checking a lot (testing in IntelliJ). It's several times slower than without, regardless what you touch. 

Also affects auto-complete suggestions. The new types allow providing suggestions for which columns you can use, but it takes several seconds to calculate them. If you just force IntelliJ to auto-complete without waiting for several seconds after typing, it just shows some random stuff.


https://github.com/vaadin/hilla/assets/357820/6553d845-e7c9-4d5d-b309-90449e28d082

Comment (platosha) [reply packages/ts/react-crud/src/autogrid.tsx:26]: I have limited inheritance to 3 levels deep, did it help?

Comment (sissbruecker) [reply packages/ts/react-crud/src/autogrid.tsx:26]: Performance is a lot better now, the auto-complete as well. Now it starts suggesting sensible options right after typing the first letter of a name.

File: packages/ts/react-crud/src/autogrid.tsx [hunk 2]
```diff
@@ -23,16 +23,63 @@ import type PropertyStringFilter from './types/dev/hilla/crud/filter/PropertyStr
...
+
+type UnsupportedColumnDataTypes = unknown[];
```
Comment (sissbruecker) [inline packages/ts/react-crud/src/autogrid.tsx:28]: The property filtering doesn't quite match what is implemented in code. It suggests nested properties from `@ManyToOne` relations which are not included by this logic here: 
https://github.com/vaadin/hilla/blob/ecc473cfabc068803a9cada7b75760c4365c6b3d/packages/ts/react-crud/src/property-info.ts#L66-L91

The same would apply for any other object model not annotated with `@OneToOne`.

If we want to make this work, we would have to:
- Include nested properties from all object models
- But do not show them by default

Comment (sissbruecker) [reply packages/ts/react-crud/src/autogrid.tsx:28]: I was using `GridWithEntityReferences.tsx` for testing.

**Comment by platosha** [comment]
As noted above, this idea is not compatible with the way nested objects work right now.  Let me close this and focus on solving #1496 instead.

**Pull request closed without merging**